pkg/plugin/connector/connutils/schema.go

```diff
@@ -0,0 +1,96 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connutils
+
+import (
+	"context"
+
+	"github.com/conduitio/conduit-connector-protocol/pconduit"
+	conduitschemaregistry "github.com/conduitio/conduit-schema-registry"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"github.com/conduitio/conduit/pkg/schemaregistry"
+	"github.com/conduitio/conduit/pkg/schemaregistry/fromschema"
+	"github.com/conduitio/conduit/pkg/schemaregistry/toschema"
+	"github.com/twmb/franz-go/pkg/sr"
+)
+
+type SchemaService struct {
+	registry schemaregistry.Registry
+	logger   log.CtxLogger
+}
+
+var _ pconduit.SchemaService = (*SchemaService)(nil)
+
+func NewSchemaService(logger log.CtxLogger, registry schemaregistry.Registry) *SchemaService {
+	return &SchemaService{
+		registry: registry,
+		logger:   logger.WithComponent("connutils.SchemaService"),
+	}
+}
+
+func (s *SchemaService) Check(ctx context.Context) error {
+	r, ok := s.registry.(schemaregistry.RegistryWithCheck)
+	if !ok {
+		return nil
+	}
+	return r.Check(ctx)
+}
+
+func (s *SchemaService) CreateSchema(ctx context.Context, req pconduit.CreateSchemaRequest) (pconduit.CreateSchemaResponse, error) {
+	ss, err := s.registry.CreateSchema(ctx, req.Subject, sr.Schema{
+		Schema: string(req.Bytes),
+		Type:   fromschema.SrSchemaType(req.Type),
+	})
+	if err != nil {
+		var respErr *sr.ResponseError
+		if cerrors.As(err, &respErr) {
+			return pconduit.CreateSchemaResponse{}, unwrapSrError(respErr) // don't wrap response errors
+		}
+		return pconduit.CreateSchemaResponse{}, cerrors.Errorf("failed to create schema: %w", err)
+	}
+	return pconduit.CreateSchemaResponse{
+		Schema: toschema.SrSubjectSchema(ss),
+	}, nil
+}
+
+func (s *SchemaService) GetSchema(ctx context.Context, req pconduit.GetSchemaRequest) (pconduit.GetSchemaResponse, error) {
+	ss, err := s.registry.SchemaBySubjectVersion(ctx, req.Subject, req.Version)
+	if err != nil {
+		var respErr *sr.ResponseError
+		if cerrors.As(err, &respErr) {
+			return pconduit.GetSchemaResponse{}, unwrapSrError(respErr) // don't wrap response errors
+		}
+		return pconduit.GetSchemaResponse{}, cerrors.Errorf("failed to get schema by subject and version: %w", err)
+	}
+
+	return pconduit.GetSchemaResponse{
+		Schema: toschema.SrSubjectSchema(ss),
+	}, nil
+}
+
+func unwrapSrError(e *sr.ResponseError) error {
+	switch e.ErrorCode {
+	case conduitschemaregistry.ErrorCodeSubjectNotFound,
+		conduitschemaregistry.ErrorCodeVersionNotFound,
+		conduitschemaregistry.ErrorCodeSchemaNotFound:
+		return pconduit.ErrSchemaNotFound
+	case conduitschemaregistry.ErrorCodeInvalidSchema:
+		return pconduit.ErrInvalidSchemaBytes // TODO change to ErrInvalidSchema
```

Comment:
Sidenote: apparently, we don't get the info from the schema registry if the bytes are invalid, the type or the subject. We should change the errors in the protocol to match this behavior.

Reply:
Can you elaborate a bit more on the change in the protocol?

Reply:
It's about the errors that are defined in the [connecor protocol](https://github.com/ConduitIO/conduit-connector-protocol/blob/35d371d9907ac9b0053b6405c514600118146e86/pconduit/errors.go#L22-L25) vs the ones in the [schema registry](https://github.com/ConduitIO/conduit-schema-registry/blob/277080c2e0462ac2c41d02bf467cc43dc39c8d65/errors.go#L43-L62). We should have the same errors (or a subset) in the protocol, since that's all the errors we get from the registry.